### PR TITLE
examples/paho-mqtt: makefile improvement and typo correction

### PR DESCRIPTION
**Pull request by javierfileiv**
### Contribution description
Regarding some comments from @Alice-For, made some makefile modifications to easily change default value and corrected a typo on main.c

With this mod, default values can be changed by CLI like this:

`DEFAULT_MQTT_USER=\"userX\" DEFAULT_MQTT_CLIENT_ID="34" DEFAULT_MQTT_PWD="Secret" make  -C examples/paho-mqtt/ `
### Issues/PRs references

Improves #12647

### CODE REVIEW

File: examples/paho-mqtt/Makefile [hunk 1]
```diff
@@ -17,6 +17,16 @@ QUIET ?= 1
...
 
+ifneq (, $(DEFAULT_MQTT_CLIENT_ID))
+    CFLAGS += -DDEFAULT_MQTT_CLIENT_ID=\"$(DEFAULT_MQTT_CLIENT_ID)\"
+endif
+ifneq (, $(DEFAULT_MQTT_USER))
+    CFLAGS += -DDEFAULT_MQTT_USER=\"$(DEFAULT_MQTT_USER)\"
+endif
+ifneq (, $(DEFAULT_MQTT_PWD))
+    CFLAGS += -DDEFAULT_MQTT_PWD=\"$(DEFAULT_MQTT_PWD)\"
+endif
```
Comment (aabadie) [inline examples/paho-mqtt/Makefile:28]: RIOT uses 2 spaces indent in Makefile.

```suggestion
ifneq (, $(DEFAULT_MQTT_CLIENT_ID))
  CFLAGS += -DDEFAULT_MQTT_CLIENT_ID=\"$(DEFAULT_MQTT_CLIENT_ID)\"
endif
ifneq (, $(DEFAULT_MQTT_USER))
  CFLAGS += -DDEFAULT_MQTT_USER=\"$(DEFAULT_MQTT_USER)\"
endif
ifneq (, $(DEFAULT_MQTT_PWD))
  CFLAGS += -DDEFAULT_MQTT_PWD=\"$(DEFAULT_MQTT_PWD)\"
endif
```

**Review by aabadie — CHANGES_REQUESTED** [review]
Ok for the typo in the main application and the change in the Makefile (except minor style issue there).

Also the commit messages should be more focused: just prefix both of them with `examples/paho-mqtt: ` instead of just `examples: ` and `makefiles: `.

**Comment by javierfileiv** [comment]
Hi @aabadie , I'd need the label to be able to run Murdock's build. Thx :)

File: examples/paho-mqtt/Makefile [hunk 2]
```diff
@@ -17,6 +17,16 @@ QUIET ?= 1
 WIFI_SSID ?= "Your_WiFi_name"
 WIFI_PASS ?= "Your_secure_password"
 
+ifneq (, $(DEFAULT_MQTT_CLIENT_ID))
```
Comment (maribu) [inline examples/paho-mqtt/Makefile:20]:
```suggestion
ifneq (,$(DEFAULT_MQTT_CLIENT_ID))
```

The space needs to be removed. More instances below.

**Review by maribu — COMMENTED** [review]
See inline comment. Also: The PR title should be changed to comply with "subsystem: description"

File: examples/paho-mqtt/Makefile (hunk 2 above)
Comment (javierfileiv) [reply examples/paho-mqtt/Makefile:20]: Thanks for your comments. Will fix it when I get some spare time 👍

**Comment by aabadie** [comment]
please squash!

**Comment by javierfileiv** [comment]
squash done!

**Review by aabadie — APPROVED** [review]
ACK!

**Comment by javierfileiv** [comment]
Thx

**Pull request merged**